frontend/components/LoadGraph.jsx

```diff
@@ -0,0 +1,144 @@
+import Chart from 'chart.js';
+import React, { createRef, useEffect } from 'react';
+import 'chartjs-plugin-zoom';
+import { Button } from '@material-ui/core';
+import RotateLeftIcon from '@material-ui/icons/RotateLeft';
+import { blue, green, orange } from '@material-ui/core/colors';
+
+var chart = null;
+const update_interval = 1000;
+const command = 'uptime';
+const chart_title = 'Load Averages';
+
+const chart_config = {
+  type: 'line',
+  data: {
+    labels: [],
+    datasets: [
+      {
+        label: '1 minute',
+        data: [],
+        backgroundColor: 'rgba(255, 100, 100, 0.1)',
+        borderColor: orange.A100,
+        borderWidth: 1.5,
+      },
+      {
+        label: '5 minute',
+        data: [],
+        backgroundColor: 'rgba(50, 255, 50, 0.1)',
+        borderColor: green.A200,
+        borderWidth: 1.5,
+      },
+      {
+        label: '15 minute',
+        data: [],
+        backgroundColor: 'rgba(100, 100, 255, 0.1)',
+        borderColor: blue.A100,
+        borderWidth: 1.5,
+      },
+    ],
+  },
+  options: {
+    title: {
+      display: true,
+      text: chart_title,
+      fontSize: 16,
+    },
+    hover: {
+      animationDuration: 0,
+    },
+    responsiveAnimationDuration: 0,
+    scales: {
+      y: {
+        beginAtZero: true,
+      },
+    },
+    elements: {
+      line: {
+        tension: 0,
+      },
+    },
+    tooltips: {
+      mode: 'x',
+    },
+    plugins: {
+      zoom: {
+        zoom: {
+          enabled: true,
+          mode: 'x',
+        },
+        pan: {
+          enabled: true,
+          mode: 'x',
+        },
+      },
+    },
+  },
+};
+
+export default function LoadGraph() {
+  const chartRef = createRef();
+
+  function updateData(obj) {
+    if (chart == null) return;
+    chart.data.labels.push(obj.x_value);
+    chart.data.datasets.forEach((dataset, idx) => {
+      dataset.data.push(obj.y_value[idx]);
+    });
+    chart.update();
+  }
+
+  function extractLoadFromString(str = '') {
+    const list = str.split(',');
+    const time_stamp = list[0].trim();
```

Comment:
This does not really work (atleast on my system). I get something like `xx:xx:xx up for x days x:xx` instead of just the `xx:xx:xx` part.

Reply:
got it